docs/celo-codebase/protocol/transactions/escrow.md

```diff
@@ -1,25 +1,224 @@
+
+    ```ts
+    function generateDeterministicInviteCode(
+        recipientPhoneHash: string,
+    recipientPepper: string,
+    addressIndex: number = 0,
+    changeIndex: number = 0,
+    derivationPath: string = CELO_DERIVATION_PATH_BASE
+    ): { privateKey: string; publicKey: string } {
+        const seed = keccak256(recipientPhoneHash + recipientPepper) as Buffer
+    return generateKeysFromSeed(seed, changeIndex, addressIndex, derivationPath)
+    }
+    ```
+
+2. converting the `public key` into a `public address` (referred to as the `paymentId` in this context) using [`publicKeyToAddress()`](https://github.com/celo-org/celo-monorepo/blob/6b6ce69fde8f4868b54abd8dd267e5313c3ddedd/packages/sdk/utils/src/address.ts#L38) from [@celo/utils/lib/address](https://github.com/celo-org/celo-monorepo/blob/6b6ce69fde8f4868b54abd8dd267e5313c3ddedd/packages/sdk/utils/src/address.ts).
+
+    ```ts
+    const publicKeyToAddress = (publicKey: string) =>
+        toChecksumAddress(
+            ensureLeading0x(pubToAddress(toBuffer(ensureLeading0x(publicKey)), true).toString('hex'))
+        )
+    ```
+
+You can find Valora's implementation of the phone number-based escrow payment in [Github > valora-inc > wallet > src > escrow](https://github.com/valora-inc/wallet/tree/2ec5767ac55197c8e97d449c2ea6479c3520859d/src/escrow) ([> saga.ts](https://github.com/valora-inc/wallet/blob/2ec5767ac55197c8e97d449c2ea6479c3520859d/src/escrow/saga.ts) and [> utils.ts](https://github.com/valora-inc/wallet/blob/2ec5767ac55197c8e97d449c2ea6479c3520859d/src/escrow/utils.ts)).
+
+<!-- Arthur todo: add link to attestation overview -->
+
+<!-- Here are also some links to learn more about attestations, phone number mappings and phone number peppers from ODIS. -->
+
+### Comparison
+
+|       | Private key-based escrow                                                        | Phone number-based escrow                                                       |
+|-------|---------------------------------------------------------------------------|---------------------------------------------------------------------------|
+| Pro ✅ | Privacy preserving (cannot be tied to a user's identity)                  | Simply requires proving ownership over phone number                       |
+| Con ❌ | Requires secretely sharing private key via SMS, messaging service or another method | Payment can be linked to a user with knowledge of their phone number |
+
+### Revoking an escrow payment
+
+The contract allows Alice (the sender) to revoke an unclaimed escrow payment she made and reclaim the funds. This is important if Bob never withdraws the escrow payment, or worse, the temporary key is lost or sent to an unintended recipient.
+
+Every escrow payment has an expiry time set by the sender upon creation. After an escrow payment has expired, Alice can revoke the payment, after which the payment is returned to her and deleted from the escrow contract.
+
```

Comment:
diagram is nice! as with other areas, would be awesome to have a snippet of example code on how to actually revoke an escrowed payment via ContractKit

Reply:
If possible here (esp. considering some of the audit feedback), can you actually clarify that the payment itself does not expiry -- expirySeconds just is for the amount of time that must pass before Alice can revoke the payment. We should make it explicit that after this time, it's still possible for Bob to withdraw the funds.

Reply:
Yes of course, great call out!